fairlearn/datasets/_fetch_diabetes_hospital.py

```diff
@@ -69,19 +73,22 @@ def fetch_diabetes_hospital(
     dataset : :obj:`~sklearn.utils.Bunch`
         Dictionary-like object, with the following attributes.
 
-        data : ndarray, shape (101766, 25)
-            Each row corresponding to the 25 feature values in order.
+        data : ndarray, shape (101766, 24)
```

Comment:
I'm a bit surprised that his changed.

Reply:
Me as well, I triple checked it. 😅 
```python
>>> data_dict['data'].values.shape
(101766, 24)
```

Reply:
Ah I think the reason is that we count the target column. So either we need to change that everywhere or leave it as n+1

Reply:
Well, if the options are "being right" and "being lazy", I will always opt for "being lazy". 😂

Reply:
Is this something to put up for discussion and create a follow-up issue for it? 🤔

Reply:
I agree with at least having a discussion. I have stumbled upon this (and over it) a couple of times, too. Let's create an issue so that people can comment.

Reply:
The new value (24) is correct. The old value (25) was incorrect. Consistently with all other datasets, we should be reporting the size of data.data, and not data.frame.

Reply:
I would suggest putting that change into a new PR, do you agree?

Reply:
It's fine either way. Are you planning to make that change, too? If not, lmk and I'll create the issue.

Reply:
I'll do it. 👍

Reply:
@romanlutz @MiroDudik each dataset has a block description in the docstring, e.g.
```python
    ==============   ====================
    Samples total                   45211
    Dimensionality                     17
    Features         numeric, categorical
    Classes                             2
    ==============   ====================
```
for the `bank_marketing` dataset. Would you count the target against the `Dimensionality` listed here, i.e. keeping the `n+1` or list the dimensionality of the feature vector?

Reply:
As @MiroDudik wrote earlier it's probably best to list the number of features, i.e., excluding the target column.

Reply:
@shimst3r -- if you are making the documentation change in this PR, can you update all three data sets that currently list n+1, so they list n? In all three data sets, the incorrect value n+1 appears in three places -- at the top of the doc, and also in the returned value section (where it appears twice).

Reply:
It's 4 places, right? Because the `feature_names` is also listing n+1 if I'm not mistaken. :)

Reply:
@MiroDudik anyway, changes are pushed. 😄